File: server/src/pharmacy/dto/pharmacy.dto.ts
import { IsDefined, IsNotEmpty, ValidateIf } from "class-validator";

export class PharmacyDto {

    @IsNotEmpty()
    readonly name: string;
    
    @ValidateIf(o => (!o.latitude || !o.longitude) || o.address)
    readonly address: string;
    
    @ValidateIf(o => !o.address || (o.latitude && o.longitude))
    readonly latitude: number;
    @ValidateIf(o => !o.address || (o.latitude && o.longitude))
    readonly longitude: number;

    readonly photoPath: string;

}

File: client/app/src/main/java/pt/ulisboa/tecnico/cmov/pharmacist/pharmacy/dto/PharmacyDto.kt
package pt.ulisboa.tecnico.cmov.pharmacist.pharmacy.dto

import kotlinx.serialization.Serializable

@Serializable
data class PharmacyDto(
    val name : String,
    val address : String?,
    val latitude : Double,
    val longitude : Double,
    var isFavorite : Boolean? = false
)
